### PR TITLE
Simplified decorators and added a simple run_eval function.

### PR DESCRIPTION
**Pull request by pjoshi30**
* Simplified the `Detect` decorator include an async and a publish parameter. This replaces the need for the `analyze_prod` decorator.
* Added a simple `run_eval` function which is meant to replace the `analyze_eval` decorator.

We need a few changes in the backend for this - a separate internal PR will be created for that change.

### CODE REVIEW

File: aimon/decorators/analyze.py [hunk 1]
```diff
@@ -16,6 +18,133 @@ def __init__(self, name, model_type, metadata={}):
...
+
+def run_eval(
+        application_name,
+        model_name,
+        dataset_collection_name, 
+        evaluation_name, 
+        headers, 
+        api_key=None,
+        aimon_client=None,
+        eval_tags=None,
```
Comment (alexlyzhov) [inline aimon/decorators/analyze.py:41]: eval_tags is not used in the code

File: aimon/decorators/analyze.py [hunk 2]
```diff
@@ -81,6 +210,11 @@ def func(context_docs, user_query, prompt, instructions *args, **kwargs):
...
         """
         super().__init__(application, model, api_key, config)
+        warnings.warn(
+            f"{self.__class__.__name__} is deprecated and will be removed in a later release. Please use the run_eval method instead.",
+            DeprecationWarning,
+            stacklevel=2
+        )
         self.headers = headers
         self.evaluation_name = evaluation_name
         self.dataset_collection_name = dataset_collection_name
```
Comment (alexlyzhov) [inline aimon/decorators/analyze.py:220]: Are eval_tags used elsewhere?

**Review by alexlyzhov — APPROVED** [review]
Left a small comment, otherwise it looks good to me!

File: aimon/decorators/analyze.py (hunk 1 above)
Comment (pjoshi30) [reply aimon/decorators/analyze.py:41]: yes, not right now. It will be good to remove it.

**Pull request merged**